cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -761,7 +761,8 @@ def _get_connection_info(self, volume, backing, connector):
             'volume': backing.value,
             'volume_id': volume.id,
             'name': volume.name,
-            'profile_id': self._get_storage_profile_id(volume)
+            'profile_id': self._get_storage_profile_id(volume),
+            'datastore': self.volumeops.get_datastore(backing).value,
```

Comment:
Will this be saved in Nova for all volume-attachments then? Where does Nova use it? Do we have to keep it up-to-date when migrating an attached volume from on DS to another like we do right now for balancing?

Reply:
> Will this be saved in Nova for all volume-attachments then?

Yes, it will be stored in the block-device-mapping of all new volume-attachments

> Where does Nova use it?

Since I added it just now, nowhere. But it is a mandatory parameter for the 
[VirtualMachineRelocateSpecDiskLocator](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/6b586ed2-655c-49d9-9029-bc416323cb22/fa0b429a-a695-4c11-b7d2-2cbc284049dc/doc/vim.vm.RelocateSpec.DiskLocator.html). Since we only relocate the (cinder)-disks on a live-migration, that would be the only sensible use-case.

> Do we have to keep it up-to-date when migrating an attached volume from on DS to another like we do right now for balancing?

No, it is only used as a target datastore for the live-migration, as soon as that is finished, the value is of no consequence.

Reply:
We might want to document this somewhere, so we can stumble upon it again when we try to find out if we have to update it in the future. Do you mind writing it into the commit-message?

Reply:
I've updated the commit message.